File: 2022/day10/cathode2.cpp
#include <iostream>
#include <fstream>
#include <list>
#include <cstring>
using namespace std;
class CPU {
    public:
        int cycleCount;
        CPU() {
            X = 2;
            clock = 0;
            cycleCount = 0;
        }
        void exec(string inst) {
            if(inst.starts_with("noop")) {
                cycle();
            }
            else {
                cycle();
                cycle();
                int val = stoi(inst.substr(5));
                X+=val;
            }
        }
        void cycle() {
            clock+=1;
            if(abs(clock%40-X)<2) {
                cout << "█";
            }
            else {
                cout << " ";
            }
            if(clock%40==0) {
                cout << "\n";
            }
        }
    private:
        int X;
        int clock;
};
int main(int argc, char *argv[]) {
    ifstream input("input.txt");
    string fileText;
    int cnt = 0;
    CPU c = CPU();
    while(getline(input, fileText)) {
        c.exec(fileText);
        cnt+=1;
    }
}

File: 2022/day10/cathode2.jl
inst = Int[]
open("input.txt") do file
    for line in eachline(file)
        if line[1:4]=="noop"
            push!(inst, 0)
        end
        if line[1:4]=="addx"
            push!(inst, 0, parse(Int,line[5:end]))
        end
    end
end
sumlist = [sum(inst[1:x]) for x in 1:length(inst)]
sumlist'
map = [abs((x-1)%40-sumlist[x])<2 for x in 1:length(sumlist)]
maped = reshape(map,40,6)'
for x in 1:6
    for y in 1:40
        if maped[x,y]
            print('█')
        else
            print(' ')
        end
    end
    println("")
end